Makefile

```diff
@@ -150,3 +169,100 @@ install-e2e-tools:
 .PHONY: print-image-tag
 print-image-tag:
 	@echo $(IMAGE)
+
+.PHONY: gsutil
+gsutil:
+	hack/install-gsutil.sh
+
+.PHONY: crossbuild-provider-aws-linux-amd64 crossbuild-provider-aws-linux-arm64
+crossbuild-provider-aws-linux-amd64 crossbuild-provider-aws-linux-arm64: crossbuild-provider-aws-linux-%:
+	mkdir -p ${DIST}/linux/$*
+	GOOS=linux GOARCH=$* go build ${GCFLAGS} ${BUILDFLAGS} ${EXTRA_BUILDFLAGS} -o ${DIST}/linux/$*/provider-aws ${LDFLAGS}"${EXTRA_LDFLAGS} -X k8s.io/provider-aws.Version=${VERSION} -X k8s.io/provider-aws.GitVersion=${GITSHA}" k8s.io/provider-aws/cmd/provider-aws
+
+.PHONY: crossbuild-provider-aws-darwin-amd64 crossbuild-provider-aws-darwin-arm64
+crossbuild-provider-aws-darwin-amd64 crossbuild-provider-aws-darwin-arm64: crossbuild-provider-aws-darwin-%:
+	mkdir -p ${DIST}/darwin/$*
+	GOOS=darwin GOARCH=$* go build ${GCFLAGS} ${BUILDFLAGS} ${EXTRA_BUILDFLAGS} -o ${DIST}/darwin/$*/provider-aws ${LDFLAGS}"${EXTRA_LDFLAGS} -X k8s.io/provider-aws.Version=${VERSION} -X k8s.io/provider-aws.GitVersion=${GITSHA}" k8s.io/provider/cmd/provider
+
+.PHONY: crossbuild-provider-aws-windows-amd64
+crossbuild-provider-aws-windows-amd64:
+	mkdir -p ${DIST}/windows/amd64
+	GOOS=windows GOARCH=amd64 go build ${GCFLAGS} ${BUILDFLAGS} ${EXTRA_BUILDFLAGS} -o ${DIST}/windows/amd64/provider-aws.exe ${LDFLAGS}"${EXTRA_LDFLAGS} -X k8s.io/provider-aws.Version=${VERSION} -X k8s.io/provider-aws.GitVersion=${GITSHA}" k8s.io/provider-aws/cmd/provider-aws
+
+.PHONY: crossbuild
+crossbuild: crossbuild-provider-aws
+
+.PHONY: crossbuild-provider-aws
+crossbuild: crossbuild-provider-aws-linux-amd64 crossbuild-provider-aws-linux-arm64 crossbuild-provider-aws-darwin-amd64 crossbuild-provider-aws-darwin-arm64 crossbuild-provider-aws-windows-amd64
+
+.PHONY: nodeup-amd64 nodeup-arm64
+nodeup-amd64 nodeup-arm64: nodeup-%:
```

Comment:
`nodeup` is a kops binary, this should be `ecr-credential-provider`, but we already have a build target for that, so what we need is a target to copy it. This is satisfied by `copy-bins-for-upload`.

Reply:
updated this part